operating_unit/models/res_users.py

```diff
@@ -74,21 +74,23 @@ def _onchange_operating_unit_ids(self):
                 record.default_operating_unit_id = False
 
     @api.depends("groups_id", "assigned_operating_unit_ids")
+    @api.depends_context("allowed_company_ids")
     def _compute_operating_unit_ids(self):
+        if self.env.context.get("allowed_company_ids"):
+            dom = [
+                "|",
+                ("company_id", "=", False),
+                ("company_id", "in", self.env.context["allowed_company_ids"]),
+            ]
+        else:
+            dom = []
         for user in self:
             if user.has_group("operating_unit.group_manager_operating_unit"):
-                dom = []
-                if self.env.context.get("allowed_company_ids"):
-                    dom = [
-                        "|",
-                        ("company_id", "=", False),
-                        ("company_id", "in", self.env.context["allowed_company_ids"]),
-                    ]
-                else:
-                    dom = []
-                user.operating_unit_ids = self.env["operating.unit"].sudo().search(dom)
+                user.operating_unit_ids = self.env["operating.unit"].search(dom)
```

Comment:
Not sure if removing the sudo is ok. That could lead to disrupts for current users?

Reply:
The function is already `compute_sudo=True`. I removed it because it seemed superfluous.

Reply:
@AaronHForgeFlow of course we will